src/sentry/api/endpoints/organization_check_has_mobile_app_events.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+
+from django.core.cache import cache
+from datetime import timedelta
+from django.utils import timezone
+
+from sentry.api.bases import OrganizationEventsEndpointBase
+from sentry.snuba import discover
+
+# one day cache
+CACHE_TTL = 24 * 60 * 60
+
+
+logger = logging.getLogger(__name__)
+
+
+class OrganizationCheckHasMobileAppEvents(OrganizationEventsEndpointBase):
+    def get(self, request, organization):
+        # cache is unique to an org
+        cache_key = f"check_mobile_app_events.{organization.id}"
+        cache_value = cache.get(cache_key)
+        # cache miss, lookup and store value
+        if cache_value is None:
+            result = self._get(request, organization)
+            # store results in an object since we don't result=None to be a cache miss
+            cache_value = {"result": result}
+            cache.set(cache_key, cache_value, CACHE_TTL)
+
+        return self.respond(cache_value["result"])
+
+    # find a match not using the cache
+    def _get(self, request, organization):
+        project_ids = self.get_requested_project_ids(request)
+        projects = self.get_projects(request, organization, project_ids)
+        if len(projects) == 0:
+            return None
+
+        browser_name_list = request.GET.getlist("userAgents")
+        query = " OR ".join(map(lambda x: f"browser.name:{x}", browser_name_list))
+
+        with self.handle_query_errors():
+            result = discover.query(
+                query=query,
+                selected_columns=["browser.name", "project", "id", "client_os_name"],
```

Comment:
I think here you want
```suggestion
                selected_columns=["browser.name", "project", "id", "client_os.name"],
```

And if the endpoint doesn't need project + id to drop them altogether so there's less data that snuba has to ask for. (Just add them back in if needed)

Reply:
@wmak would like them for logging so I can debug some of the events and load them up in Sentry.

Reply:
d'oh I even suggested using the `**dict` syntax, :+1: Makes sense to keep them then